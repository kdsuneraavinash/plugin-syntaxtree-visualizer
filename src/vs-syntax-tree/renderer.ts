import { ExtendedLangClient } from '../core/extended-language-client';
import { ExtensionContext } from 'vscode';
import { getLibraryWebViewContent, WebViewOptions, getComposerWebViewOptions } from '../utils';

export function render(context: ExtensionContext, langClient: ExtendedLangClient, sourceRoot: string)
    : string {

    const body = `
            <div class="container">
                <div id="treeBody" />
            </div>
    `;
    const bodyCss = ``;
    const styles = `
        .container {
            overflow-x: auto;
            position: relative;
            height: 100%;
            text-align: center;
            padding-top: 2%;
        }

        #treeBody {
            display: inline-block;
            position: relative;
        }
    `;
    const scripts = `
        function loadedScript() {
            let docUri = ${JSON.stringify(sourceRoot)};
            let collapsedNode = "";
            let isGraphical = false;

            window.addEventListener('message', event => {
                let msg = event.data;
                switch(msg.command){
                    case 'update':
                        docUri: msg.docUri;
                        initiateRendering();
                }
            });

            function renderTree(){
                return new Promise((resolve, reject) => {
                    webViewRPCHandler.invokeRemoteMethod('fetchSyntaxTree', [docUri], (response) => {
                        if(!response.parseSuccess){
                            document.getElementById("treeBody").innerHTML = "Oops! Something went wrong!";
                        }
                        else {
                            webViewRPCHandler.invokeRemoteMethod('fetchTreeGraph', [response], (result) => {
                                resolve(result);
                            });
                        }
                    });
                })
            }

            function collapseTree(nodeID, representationType){
                collapsedNode = nodeID;
                isGraphical = representationType;
                ballerinaComposer.renderSyntaxTree(collapseTree, collapseNodes, document.getElementById("treeBody"));
            }

            function collapseNodes(){
                return new Promise((resolve, reject) => {
                    webViewRPCHandler.invokeRemoteMethod('onCollapseTree', [collapsedNode, isGraphical], (response) => {
                        resolve(response);
                    });
                })
            }

            function initiateRendering(){
                ballerinaComposer.renderSyntaxTree(collapseTree, renderTree, document.getElementById("treeBody"));
            }

            initiateRendering();
        }
    `;

    const webViewOptions: WebViewOptions = {
        ...getComposerWebViewOptions(),
        body, scripts, styles, bodyCss
    };
    
    return getLibraryWebViewContent(webViewOptions);
}
