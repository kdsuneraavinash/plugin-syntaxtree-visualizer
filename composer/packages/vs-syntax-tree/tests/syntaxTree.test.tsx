import React from "react";
import renderer from "react-test-renderer";
import SyntaxTree from "./../src/syntaxTree";

it("Syntax tree renders correctly", () => {
  const tree = renderer
    .create(<SyntaxTree renderTree = {jest.fn()} onCollapseTree = {jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
