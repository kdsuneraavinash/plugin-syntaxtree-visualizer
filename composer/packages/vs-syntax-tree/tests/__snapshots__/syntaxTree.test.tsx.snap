// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Syntax tree renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "fontSize": "15px",
      }
    }
  >
    <p
      style={
        Object {
          "paddingRight": "15px",
        }
      }
    >
      Graphical Tree View
    </p>
    <div
      className="ui fitted toggle checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        onClick={[Function]}
        readOnly={true}
        tabIndex={0}
        type="radio"
      />
      <label />
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "position": "relative",
      }
    }
  >
    <div
      className="ui inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui medium text loader"
        >
          Loading
        </div>
      </div>
    </div>
  </div>
</div>
`;
